### DISCUSSION TITLE
VolumeSnapshotClass is not applying on k8s 1.21

### DISCUSSION THREAD

**Discussion by darvish-git** [rook/rook#8148] [Q&A] [answered]
rook-volumesnapshotclass.yaml": no matches for kind "VolumeSnapshotClass" in version "snapshot.storage.k8s.io/v1"
I saw commits which fixed  this with change vbetav1 to v1. But it is not working in my env
k8s: 1.21
OS : centos 8
docker: 20.10.6
rook: 15.2.9

**Comment by Madhu-1** [comment, marked as answer]
Have you verified required per-req mentioned at https://github.com/rook/rook/blob/master/Documentation/ceph-csi-snapshot.md#prerequisites

**Reply by darvish-git** [reply]
Yes. Everything same like documentation

**Reply by Madhu-1** [reply]
From the above error it's looks like snapshot CRD is not installed can you run `kubectl get crd| grep -i snapshot` if you are able to see snapshot check it's v1 version

**Reply by darvish-git** [reply]
Above I wrote error which can't install  ceph/csi/rbd/snapshotclass.yaml.  It needs for backup and dissaster recovery for rook+ceph with kasten10. Now I can't install rbd/snapshotclass.yaml on k8s v1.21

**Reply by darvish-git** [reply]
with storageclass.yaml everything ok.

**Reply by darvish-git** [reply]
and for information : validationwebhook api version is v1.

**Comment by Madhu-1** [comment]
check `Install the snapshot controller and snapshot v1/v1beta1 CRD as required.` on pre-req  https://github.com/rook/rook/blob/master/Documentation/ceph-csi-snapshot.md#prerequisites . Rook won't install snapshot CRD the admin has to install the snapshot CRD from the external-snapshotter repo or it should come with kubernetes distrubution.

**Reply by darvish-git** [reply]
> check `Install the snapshot controller and snapshot v1/v1beta1 CRD as required.` on pre-req https://github.com/rook/rook/blob/master/Documentation/ceph-csi-snapshot.md#prerequisites . Rook won't install snapshot CRD the admin has to install the snapshot CRD from the external-snapshotter repo or it should come with kubernetes distrubution.

Thank for help. It's my mistake. Didn't read top requirement on docs. CRD fixed error
